packages/medusa/src/strategies/cart-completion.ts

```diff
@@ -47,72 +70,77 @@ class CartCompletionStrategy implements ICartCompletionStrategy {
     while (inProgress) {
       switch (idempotencyKey.recovery_point) {
         case "started": {
-          const { key, error } = await idempotencyKeyService.workStage(
-            idempotencyKey.idempotency_key,
-            async (manager: EntityManager) => {
-              const cart = await cartService
-                .withTransaction(manager)
-                .retrieve(id)
-
-              if (cart.completed_at) {
-                return {
-                  response_code: 409,
-                  response_body: {
-                    code: MedusaError.Codes.CART_INCOMPATIBLE_STATE,
-                    message: "Cart has already been completed",
-                    type: MedusaError.Types.NOT_ALLOWED,
-                  },
+          const { key, error } = await idempotencyKeyService
+            .withTransaction(this.manager_)
+            .workStage(
+              idempotencyKey.idempotency_key,
+              async (manager: EntityManager) => {
+                const cart = await cartService
```

Comment:
thought: Sorry for intruding on your Draft PR but just wanted to highlight that each `case-block` should be executed in its own transaction so if we use pass the same `this.manager_` to each of these we are arriving at the same issue as we discussed earlier; namely that there are "nested transactions" which we want to avoid when using idempotency keys. 

I think it might not make sense at all to make strategies inherit extend the transaction service. Instead, let's consider making all strategies registered in the container into transient registrations instead. Wdyt?

Reply:
Yeah indeed, i planed to revert to the previous approach i did, but tomorrow, which is the one we applied in other places with each block in his own transaction. I just went back to this one for no reason 😂 i think just memory muscle 💪, also, i think that some strategies could still required to be wrap in a transaction but it is a case by case choice. Some strategies could interact with the database to create/update/read some objects and therefore could need to be in a transaction in case we are trying to access the same data that we are also manipulating in the core.

For this strategy, being scoped should suffice no? And then we can think of the scope of each strategy independently since i don’t think all of them need to be re instanciate all the time. Wdyt?